app/src/common/shared/com/igalia/wolvic/speech/SpeechRecognizer.java

```diff
@@ -40,4 +40,7 @@ interface Callback {
     boolean shouldDisplayStoreDataPrompt();
     default boolean supportsASR(@NonNull Settings settings) {return true;}
     boolean isSpeechError(int code);
+    String[] getSupportedLanguagesNames();
+    int getIndexForLanguageId(String languageId);
+    String getLanguageIdForIndex(int checkedRadioButtonId);
```

Comment:
I think we could live with it temporarily because it follows the same structure as the current code but I strongly disagree with having these 2 methods in the speech recognizers. Those two methods are used to associate languages with visual elements (radio buttons in this case). This should be something that  must be done at a very different (higher) level. The settings widget should be the one getting the list of supported languages and building and controlling the UI to select the language instead of relying on some indexes from some random array in the speech recognizers. 

Please file a bug report and do add a FIXME linking to that issue here so that we don't forget,

Reply:
Good point. I opened https://github.com/Igalia/wolvic/issues/151